src/static/js/language.js

```diff
@@ -0,0 +1,42 @@
+const supportedLang = ['en', 'ja'];
+
+const langSwitcher = document.querySelector('.language-switcher');
+
+langSwitcher.addEventListener('change', (e) => {
+  if (e.target.value) {
+    window.location.href =
+      updatePath(window.location.pathname, e.target.value);
+  }
+});
+
+function updatePath(path, value) {
+  // Current path has no lang prefix
+  if (path.length === 1) {
+    return path + value;
+  }
+
+  const langPrefix = path.substring(1,3);
+  if (supportedLang.includes(langPrefix) && path[3] === '/') {
+    return path.replace(langPrefix, value);
+  }
+}
+
+const defaultOption = document.querySelector('.default-language');
+const alternateOptions = document.querySelectorAll('.alternate-language');
+
+// Generate default language option
+defaultOption.innerHTML = generateOptionText(defaultOption.value);
```

Comment:
You can update all of the options at the same time, so this special case can be removed.